Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,500 @@
+                                               CREDIT_MINING_FOLDER_DOWNLOAD)
+        self.load_config = load_config
+
+
+class BoostingManager(TaskManager):
+    """
+    Class to manage all the credit mining activities
+    """
+
+    __single = None
+
+    def __init__(self, session, settings=None):
+        super(BoostingManager, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        BoostingManager.__single = self
+        self.boosting_sources = {}
+        self.torrents = {}
+
+        self.session = session
+        assert self.session.get_libtorrent()
+
+        self.torrent_db = self.session.open_dbhandler(NTFY_TORRENTS)
+        self.channelcast_db = self.session.open_dbhandler(NTFY_CHANNELCAST)
+
+        # use provided settings or a default one
+        self.settings = settings or BoostingSettings(session, load_config=True)
+
+        if self.settings.load_config:
+            self._logger.info("Loading config file from session configuration")
+            self.load_config()
+
+        if not os.path.exists(self.settings.credit_mining_path):
+            os.makedirs(self.settings.credit_mining_path)
+
+        self.session.lm.ltmgr.get_session().set_settings(
+            {'share_mode_target': self.settings.share_mode_target})
+
+        self.session.add_observer(self.on_torrent_notify, NTFY_TORRENTS, [NTFY_UPDATE])
+
+        self.register_task("CreditMining_select", LoopingCall(self._select_torrent),
+                           self.settings.initial_swarm_interval, interval=self.settings.swarm_interval)
+
+        self.register_task("CreditMining_scrape", LoopingCall(self.scrape_trackers),
+                           self.settings.initial_tracker_interval, interval=self.settings.tracker_interval)
+
+        self.register_task("CreditMining_log", LoopingCall(self.log_statistics),
+                           self.settings.initial_logging_interval, interval=self.settings.logging_interval)
+
+    @staticmethod
+    def get_instance(*args, **kw):
+        """
+        get single instance of Boostingmanager
+        """
+        if BoostingManager.__single is None:
+            BoostingManager(*args, **kw)
+        return BoostingManager.__single
+
+    @staticmethod
+    def del_instance():
+        """
+        resetting, then deleting single instance
+        """
+        BoostingManager.__single = None
+
+    def shutdown(self):
+        """
+        Shutting down boosting manager. It also stops and remove all the sources.
+        """
+        self.save_config()
+        self._logger.info("Shutting down boostingmanager")
+
+        for sourcekey in self.boosting_sources.keys():
+            self.remove_source(sourcekey)
+
+        self.cancel_all_pending_tasks()
+
+        #remove credit mining data in not persistent mode
+        shutil.rmtree(self.settings.credit_mining_path, ignore_errors=True)
```

Comment:
In that case, why not make a persistent mode attribute in BoostingManager and still wrap this inside an if-statement that checks this persistent mode attribute? I think it's confusing right now since the comment is not in sync with the code.